overlord/exportstate/manifest.go

```diff
@@ -0,0 +1,209 @@
+
+/*
+ * Copyright (C) 2020-2022 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package exportstate
+
+import (
+	"os"
+	"path/filepath"
+
+	"github.com/snapcore/snapd/dirs"
+	"github.com/snapcore/snapd/release"
+	"github.com/snapcore/snapd/snap"
+)
+
+// Manifest describes content content exported to snaps or the host.
+type Manifest struct {
+	SnapInstanceName          string        `json:"snap-instance-name"`
+	SnapRevision              snap.Revision `json:"snap-revision"`
+	ExportedForSnapdAsVersion string        `json:"exported-for-snapd-as-version,omitempty"`
+
+	// SourceIsHost is only true if provider of the manifest is not a snap but the classic system.
+	// All SourcePath fields, as visible through Sets[*].Exports[*].SourcePath, are absolute names
+	// in the host file system.
+	SourceIsHost bool `json:"source-is-host,omitempty"`
+
+	// Sets describe groups of exported files.
+	Sets map[string]ExportSet `json:"sets"`
+}
+
+// ExportSet describes a group of files for a given type of consumer.
+type ExportSet struct {
+	Name string `json:"name"`
+	// ConsumerIsHost is true if an export set provides files usable from the host system.
+	ConsumerIsHost bool                    `json:"consumer-is-host,omitempty"`
+	Exports        map[string]ExportedFile `json:"exports"`
+}
+
+// ExportedFile describes a single file exported from one place to another.
+//
+// Currently all files are exported as symbolic links. If this changes a new
+// field should be added here, to encode this information.
+type ExportedFile struct {
+	// Name is the name of the exported file, usually the same as the leaf component of SourcePath.
+	Name string `json:"name"`
+	// SourcePath is the path of the exported file relative to its source. The
+	// ultimate interpretation of this field depends on Manifest.SourceIsHost
+	// and ExportSet.ConsumerIsHost fields.
+	SourcePath string `json:"source-path"`
+}
+
+// NewManifestForSnap returns the manifest of a given snap or nil if there is
+// no manifest.
+//
+// Currently only the core and snapd snaps export any content to the system. As
+// such the export manifest is not embedded into the snap meta-data but instead
+// computed here.
+//
+// Both snapd and core snaps export content under the snap name "snapd", using
+// the export set name "tools". The revision is mangled, for "snapd" it is used
+// directly. For "core" it is transformed to "core_$revision".
```

Comment:
I'm not sure whether this is the best place for this comment; at least, I cannot relate it with the function below.

Edit: after looking at the rest of the code, I suggest moving it above, where you define the Manifest type and specifically the `ExportedForSnapdAsVersion` field.